http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala

```diff
@@ -237,17 +241,29 @@ class ProxyRoutes(proxyService: ProxyService, corsSupport: CorsSupport, refererC
     for {
       ctx <- ev.ask[AppContext]
       apiCall = proxyService.proxyRequest(userInfo, googleProject, runtimeName, request)
-      _ <- metrics.incrementCounter("proxyRuntime")
       resp <- ctx.span.fold(apiCall)(span => spanResource[IO](span, "proxyRuntime").use(_ => apiCall))
+
+      headerMap: Map[String, String] = request.headers.map(header => (header.name(), header.value())).toMap
```

Comment:
I think this could be a bit more concise:
```
_ <- request.header[`Content-Length`].traverse(h => metrics.gauge("proxy/notebooksSize", h.length))
```

Reply:
IntelliJ is telling me no on using the `.length`, apparently it returns a Long but the `metric.gauge` requires a Double and so it is saying using `toDouble` is necessary (which then is going bring us back to the issue Justin brought up) so I think I'll have to leave it a little more complicated

Reply:
Ok. Still might be better to use the built-in header lookup method instead of iterating through the list though. Also I think we can only do this when necessary:
```
_ <- if (request.uri.path.endsWith("*.ipynb") && request.method == HttpMethods.PUT) {
  val contentLengthOpt = request.header[`Content-Length`].flatMap(h => Try(h.length.toDouble).toOption)
  contentLengthOpt.traverse { c =>
    metrics.gauge("proxy/notebooksSize", c)
  }
} else IO.unit
```